### DISCUSSION TITLE
Linking arguments after loading a data module?

### DISCUSSION THREAD

**Discussion by rohanshad** [Lightning-AI/pytorch-lightning#15613] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I have a lightning module (model) that needs an argument to come from the DataModule (eg. batch_size). This works great when I explicitly define the data module that needs to be used: 

```python
class demo_cli(LightningCLI):
     def add_arguments_to_parser(self, parser):
            parser.link_arguments("data.batch_size", "model.batch_size")

cli = demo_cli(LightningModule, DataModule) 
```
If I however have multiple data modules, and attempt this, there appears to be no way to link arguments from DataModules loaded dynamically via the command line / config files as described in this [section of the docs](https://pytorch-lightning.readthedocs.io/en/latest/cli/lightning_cli_advanced_3.html#multiple-models-and-or-datasets). Any ideas?
```python
cli = demo_cli(LightningModule, subclass_mode_data=True) 
```

**Comment by mauvilsa** [comment, marked as answer]
The `link_argument`'s source and target keys follow the same structure as the config files. If only the data is in subclass mode, not the model, then the link would be as:

```python
parser.link_arguments("data.init_args.batch_size", "model.batch_size")
```

**Reply by rohanshad** [reply]
Perfect, worked flawlessly.
